packages/nitro-protocol/contracts/NitroAdjudicator.sol

```diff
@@ -184,6 +184,29 @@ contract NitroAdjudicator is Adjudicator, ForceMove {
 
         }
     }
+
+    /**
+    * @notice Check that the submitted pair of states form a valid transition (public wrapper for internal function _requireValidTransition)
+    * @dev Check that the submitted pair of states form a valid transition (public wrapper for internal function _requireValidTransition)
+    * @param nParticipants Number of participants in the channel.
+    transition
+    * @param isFinalAB Pair of booleans denoting whether the first and second state (resp.) are final.
+    * @param ab Variable parts of each of the pair of states
+    * @param turnNumB turnNum of the later state of the pair.
+    * @param appDefinition Address of deployed contract containing application-specific validTransition function.
+    * @return true if the later state is a validTransition from its predecessor, reverts otherwise.
+    */
+    function validTransition(
+        uint256 nParticipants,
+        bool[2] memory isFinalAB, // [a.isFinal, b.isFinal]
+        ForceMoveApp.VariablePart[2] memory ab, // [a,b]
+        uint256 turnNumB,
+        address appDefinition
+        ) public pure returns (bool) {
+
+        return _requireValidTransition(nParticipants, isFinalAB, ab, turnNumB, appDefinition);
+
+    }
```

Comment:
This will be more expensive (gas wise) than the function it wraps, due to it being a ` public` rather than `internal` function: that is, if this function ever gets called as a part of a transaction (`eth_call` is free). 

All of the transactions that a SC wallet might use still make use only of the wrapped function, so gas consumption of those transactions will be unaffected.

Reply:
We should discuss a strategy for tracking deployment addresses for each update to @statechannels/nitro-protocol.